boa_engine/src/builtins/async_generator/mod.rs

```diff
@@ -527,7 +527,7 @@ impl AsyncGenerator {
         drop(generator_context_mut);
 
         // 8. Assert: result is never an abrupt completion.
-        assert!(result.is_ok());
+        assert!(!result.is_throw_completion());
```

Comment:
`Return` is also abrupt, so `is_throw_completion` should probably be `is_abrupt_completion`

Reply:
Right now, the main use of `CompletionRecord::Return` in this PR is to signal a `Yield`, which would typically be a normal completion. So in this essence, `Throw` is the only true `abrupt completion` currently.

I did try to make the changes, but it started to break more tests in the test262 suite, which is a bit hard to determine whether it is related to `CompletionRecord` semantics that we aren't accounting for or `Generator` and `AsyncGenerator` execution.

Reply:
Hm, is that ideal? Or should we try to align with the spec?
If this is the ideal behaviour, then we can just merge this as it is, but if it's not then it would be good to open an issue for it.

Reply:
I think the overarching structure here is ideal/working towards it. But if we're referring to the current implementation of `Yield`, then I'm not 100% sure. This mostly begins to implement `CompletionRecord` while maintaining the current behavior of the VM (i.e., returning `ReturnType::Normal` and separate flag for `ReturnType::Yield`).

Reply:
Then let's just merge this and open an issue called "reevaluate the execution semantics of `CompletionRecord`" or something.